pkg/kv/kvserver/spanset/batch.go

```diff
@@ -1011,10 +1036,25 @@ func DisableUndeclaredSpanAssertions(rw storage.ReadWriter) storage.ReadWriter {
 // function.
 func DisableForbiddenSpanAssertions(rw storage.ReadWriter) storage.ReadWriter {
 	switch v := rw.(type) {
+	case *storage.OpLoggerBatch:
+		// OpLoggerBatch embeds a storage.Batch. Recursively process the inner
+		// batch and wrap it back in an OpLoggerBatch to preserve the chain.
+		innerBatch := DisableForbiddenSpanAssertions(v.Batch).(storage.Batch)
+		return v.ShallowCopyWithBatch(innerBatch)
+
 	case *spanSetBatch:
 		newSnapSetBatch := v.shallowCopy()
 		newSnapSetBatch.spans.DisableForbiddenSpansAssertions()
+
+		// Recursively disable on the underlying batch in case there are
+		// nested spanSetBatches.
+		newSnapSetBatch.b = DisableForbiddenSpanAssertions(v.b).(storage.Batch)
+		// Update the reader and writer to point to the recursively processed batch.
+		newSnapSetBatch.spanSetReader.r = newSnapSetBatch.b
+		newSnapSetBatch.spanSetWriteBatch.spanSetWriter.w = newSnapSetBatch.b
+		newSnapSetBatch.spanSetWriteBatch.wb = newSnapSetBatch.b
```

Comment:
unactionable, just comment: Took me a few minutes to comprehend this block. Probably there is a better abstraction (e.g. as suggested in previous PR could use a `make` func instead of a clone-and-modify concept).